src/App.tsx

```diff
@@ -15,36 +17,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+ enum SortType {
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+  default = '',
+}
+
+type SortObj = {
+  sortField: SortType;
+  reverseGood?: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverseGood = false }: SortObj) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+        case SortType.byLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseGood) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [reverseGood, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverseGood,
+  });
+
+  const reset = () => {
```

Comment:
The function `reset` should be named to reflect its action more accurately, following the [CODE STYLE] guideline. A name like `handleReset` would be more descriptive of its purpose.